### DISCUSSION TITLE
Reconnect to a plugged in device

### DISCUSSION THREAD

**Discussion by brianreinhold** [pyusb/pyusb#451] [Q&A] [answered]
Is it possible to reconnect to a USB device that remains physically plugged in after it has been disconnected?

Similarly, if the device is plugged in but no application connects to it, can an application connect to it sometime later?

**Comment by jonasmalacofilho** [comment, marked as answer]
> Is it possible to reconnect to a USB device that remains physically plugged in after it has been disconnected?

The answer is somewhere between "technically, yes, but..." and "sometimes".

Basically, and as far as I know, this requires getting the (potentially root) hub to reset the corresponding port.  Which at the very least requires you to issue the correct control transfer _to the hub,_ but also that the feature is available to start with; if it's a root hub (and again, AFAIK), it might not be.

I also wouldn't be surprised if there are platform differences/limitations around this (but this part is just speculation, I haven't looked into it).

> Similarly, if the device is plugged in but no application connects to it, can an application connect to it sometime later?

Yes and that's the normal case, no?

**Reply by brianreinhold** [reply]
I think in this case the application gets what it wants and simply terminates without making the effort to disconnect from the device. Thus I assume the (Raspberry PI) OS still thinks the device is connected. Doing some testing that appears to be the case, though I have not yet done a full blown test where I go further than reading the descriptors and getting the endpoints. I have not tried to actually read and write.

**Reply by jonasmalacofilho** [reply]
I agree, from what you describe the device would never not be present to the OS (but it might no longer bound to the same kernel driver, in some cases).

**Reply by jonasmalacofilho** [reply]
P.S. I had originally understood "disconnect" to mean some type of software or hardware disconnection from the host system.  In PyUSB/LibUSB it's more common, IMO, to use the terms open/close for the management of the "connection" between an application and the device.
